### PR TITLE
integration ADOP(3p) <> amp-ad

### PR DESCRIPTION
**Pull request by chojunho**
- upload ADOP Ads module

### CODE REVIEW

File: ads/adop.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright 2017 The AMP HTML Authors. All Rights Reserved.
```
Comment (torch2424) [inline ads/adop.js]: Incorrect copyright year 😄

File: ads/adop.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+<!---
+Copyright 2017 The AMP HTML Authors. All Rights Reserved.
```
Comment (torch2424) [inline ads/adop.md]: Incorrect copyright year 😄

File: ads/adop.md [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+## Configuration
+
+For configuration details and to generate your tags, please contact http://adop.cc
+
+### Required parameters
+
+- `data-z`
+
+### Optional parameters
```
Comment (torch2424) [inline ads/adop.md]: If there are no optional parameters, can we remove this header? 😄

Comment (chojunho) [reply ads/adop.md]: Sure. Thank you.

File: ads/adop.md (hunk 2 above)
Comment (chojunho) [reply ads/adop.md]: Do i have to recommit about it?

File: ads/adop.js (hunk 1 above)
Comment (chojunho) [reply ads/adop.js]: Do i have to recommit about it?

**Comment by chojunho** [comment]
Hello. I update files you requested.
Please check the files if they are right.

**Comment by torch2424** [comment]
@chojunho 

Thanks for making the changes! 😄 🎉 

So I went ahead and tested your `renderStartImplemented: true`, by editing the `amp-ad-xorigin-iframe-handler`.

Screen Shot 2019-05-24 at 4 26 30 PM
Screen Shot 2019-05-24 at 4 26 40 PM

So I added an extra console.log somewhere else, but I noticed that we never get the render start message. Is your server set up for this? 

If so, could you look into this, or am I doing something wrong?

If not, could you please implement this? It helps us serve ads as fast as possible to our users, and gives us some big performance gains. 😄 

Thanks! 👍

**Comment by chojunho** [comment]
Hello.
I worked it based on other networks module.
So unfortunately I don't understand renderStartImplemented option perfectly.
Sorry for it.
If our ad server doesn't provide it, do i remove this option on _config.js?

**Comment by chojunho** [comment]
> @chojunho
> 
> Thanks for making the changes! 😄 🎉
> 
> So I went ahead and tested your `renderStartImplemented: true`, by editing the `amp-ad-xorigin-iframe-handler`.
> 
> Screen Shot 2019-05-24 at 4 26 30 PM
> Screen Shot 2019-05-24 at 4 26 40 PM
> 
> So I added an extra console.log somewhere else, but I noticed that we never get the render start message. Is your server set up for this?
> 
> If so, could you look into this, or am I doing something wrong?
> 
> If not, could you please implement this? It helps us serve ads as fast as possible to our users, and gives us some big performance gains. 😄
> 
> Thanks! 👍

Hello.
I worked it based on other networks module.
So unfortunately I don't understand renderStartImplemented option perfectly.
Sorry for it.
If our ad server doesn't provide it, do i remove this option on _config.js?

**Comment by chojunho** [comment]
@torch2424 

Hello.
I worked it based on other networks module.
So unfortunately I don't understand renderStartImplemented option perfectly.
Sorry for it.
If our ad server doesn't provide it, do i remove this option on _config.js?

**Comment by torch2424** [comment]
@chojunho 

Ah no worries!

"If our ad server doesn't provide it, do i remove this option on _config.js?"

And yes please, if you can remove the `renderStartImplemented` in this case, I think that would be great 😄 After that I think we can merge this in.

**Comment by chojunho** [comment]
@torch2424 

Hello. I remove the option.
We hope to see our ads served by <amp-ad> as soon as possible.
Thank you for your help.

**Comment by torch2424** [comment]
"Hello. I remove the option.
We hope to see our ads served by amp-ad as soon as possible.
Thanks for your help."

Now that this is merged, it should be in PROD in about two weeks 😄 And you are welcome! 👍 

"And may i ask your slack ID?
As you know, Our amp-ad module is basic of basic version.
And I'm also interested in improving AD speed and performance.
When i will work it, I'd like to get some help."

Feel free to ask for help anytime! I'm torch2424 on Slack as well!

Thanks for the contribution again! 😄

**Pull request merged**